nexus/mgs-updates/src/rot_updater.rs

```diff
@@ -203,23 +210,218 @@ impl SpComponentUpdateHelper for ReconfiguratorRotUpdater {
+                (
+                    ExpectedVersion::Version(artifact_version),
+                    FoundVersion::Version(found_version),
+                ) if artifact_version.to_string() == *found_version => (),
+                // anything else is a mismatch
+                (ExpectedVersion::NoValidVersion, FoundVersion::Version(_))
+                | (ExpectedVersion::Version(_), FoundVersion::MissingVersion)
+                | (ExpectedVersion::Version(_), FoundVersion::Version(_)) => {
+                    return Err(PrecheckError::WrongInactiveVersion {
+                        expected: expected_inactive_slot_version.clone(),
+                        found: found_version,
+                    });
+                }
+            };
+
+            let (
+                active, pending_persistent_boot_preference, transient_boot_preference
+            ) = match &state.rot {
+                RotState::V2 {
+                    active,
+                    pending_persistent_boot_preference,
+                    transient_boot_preference,
+                    ..
+                }
+                | RotState::V3 {
+                    active,
+                    pending_persistent_boot_preference,
+                    transient_boot_preference,
+                    ..
+                } => (
+                    active, pending_persistent_boot_preference, transient_boot_preference
+                ),
+                RotState::CommunicationFailed { message } => {
+                    return Err(PrecheckError::RotCommunicationFailed { message: message.to_string() })
+                },
+            };
+
+            // If the active slot does not match the expected active slot, there is
+            // likely another update happening. Bail out.
+            if expected_active_slot != active {
+                return Err(PrecheckError::WrongActiveSlot {
+                    expected: *expected_active_slot, found: *active
+                })
+            }
+
+            // If transient boot is being used, the persistent preference is not going to match 
+            // the active slot. At the moment, this mismatch can also mean one of the partitions
+            // had a bad signature check. We don't have a way to tell this appart yet.
+            // https://github.com/oxidecomputer/hubris/issues/2066
+            //
+            // For now, this discrepancy will mean a bad signature check. That's ok, we can continue.
+            // The logic here should change when transient boot preference is implemented.
+            if expected_persistent_boot_preference != active {
+                info!(log, "expected_persistent_boot_preference does not match active slot. \
+                This could mean a previous broken update attempt.");
+            };
+
+            // If pending_persistent_boot_preference or transient_boot_preference is/are some,
+            // then we need to wait, an update is happening.
```

Comment:
I don't think the update code currently sets `transient_boot_preference`, right? Should it?

Reply:
No, it's not necessary yet. The `transient_boot_preference` functionality has not been implemented yet.